packages/material-ui/src/Menu/Menu.js

```diff
@@ -29,92 +28,92 @@ export const styles = {
   },
 };
 
-class Menu extends React.Component {
-  menuListActionsRef = React.createRef();
+const Menu = React.forwardRef(function Menu(props, ref) {
+  const {
+    children,
+    classes,
+    disableAutoFocusItem,
+    MenuListProps,
+    onClose,
+    onEntering,
+    open,
+    PaperProps = {},
+    PopoverClasses,
+    theme,
+    ...other
+  } = props;
+  const menuListActions = React.useRef();
+
+  const focus = () => {
```

Comment:
You could initialize the actions with an object that has the same interface and implement `focus` with a no-op. Then you could always call `menuListActions.current.focus` without defensive checks.